### PR TITLE
Hanna melnyk w2 java script

### PR DESCRIPTION
**Pull request by hanna-melnyk**
HANNA-MELNYK-w2-JavaScript

### CODE REVIEW

File: 1-JavaScript/Week2/assignment/ex1-giveCompliment.js [hunk 1]
```diff
@@ -18,8 +18,11 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/1-Java
    Use `console.log` each time to display the return value of the
    `giveCompliment` function to the console.
 -----------------------------------------------------------------------------*/
-function giveCompliment(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function giveCompliment(name) {
+  const compliments = ["showstopping", "spectacular", "never the same", "unique", "fantastic", "one of a kind", "breathtaking", "unparalleled", "phenomenal", "extraordinary"];
+  const randomIndex = Math.floor(Math.random() * compliments.length);
+  const compliment = compliments[randomIndex];
+  return `You are ${compliment}, ${name}!`;
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex1-giveCompliment.js:25]: 💯

File: 1-JavaScript/Week2/assignment/ex2-dogYears.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ calculate it!
    ages.
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function calculateDogAge(puppyAge) {
+  return`Your doggie is ${puppyAge * 7} years old in dog years!`
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex2-dogYears.js:20]: 💯

File: 1-JavaScript/Week2/assignment/ex3-tellFortune.js [hunk 3]
```diff
@@ -33,29 +33,35 @@ body, this code is now written once only in a separated function.
 
 // This function should take an array as its parameter and return
 // a randomly selected element as its return value.
-function selectRandomly(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function selectRandomly(elements) {
+  const randomIndex = Math.floor(Math.random() * elements.length);
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex3-tellFortune.js:37]: 💯

File: 1-JavaScript/Week2/assignment/ex3-tellFortune.js [hunk 4]
```diff
@@ -33,29 +33,35 @@ body, this code is now written once only in a separated function.
...
+  return elements[randomIndex];
 }
 
-function tellFortune(/* TODO add parameter(s) here */) {
-  // TODO complete this function
+function tellFortune(numKids, partnerNames, locations, jobTitles) {
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex3-tellFortune.js:41]: 💯

File: 1-JavaScript/Week2/assignment/ex3-tellFortune.js (hunk 3 above)
Comment (DanielGordon1) [reply 1-JavaScript/Week2/assignment/ex3-tellFortune.js:37]: We could optionally make this function a one-liner, but thats no necessity.

File: 1-JavaScript/Week2/assignment/ex4-shoppingCart.js [hunk 5]
```diff
@@ -23,8 +23,16 @@ you have more than 3 items in your shopping cart the first item gets taken out.
 const shoppingCart = ['bananas', 'milk'];
 
 // ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(item) {
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex4-shoppingCart.js:26]: Nailed it 🥇

File: 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js [hunk 6]
```diff
@@ -16,8 +16,14 @@ it pure. Do the following:
 5. Confirm that you function passes the provided unit tests.
 ------------------------------------------------------------------------------*/
 // ! Function under test
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function addToShoppingCart(cart, item) {
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js:19]: What happens if the shopping cart has 4 items in it already? Does this function still work ? 🤔

File: 1-JavaScript/Week2/assignment/ex6-totalCost.js [hunk 7]
```diff
@@ -21,22 +21,32 @@ instead!
...
+  heineken: 2.50,
+  stroopwafels: 3.99,
+  goudaCheese: 5.45,
+  herring: 4.50,
+  stamppot: 7.00
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
-  // TODO replace this comment with your code
+function calculateTotalPrice(cart) {
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex6-totalCost.js:31]: 💯

File: 1-JavaScript/Week2/assignment/ex6-totalCost.js [hunk 8]
```diff
@@ -21,22 +21,32 @@ instead!
...
+    total += cart[item];
+  }
+  return `Total: €${total.toFixed(2)}`;
 }
 
 // ! Test functions (plain vanilla JavaScript)
 function test1() {
   console.log('\nTest 1: calculateTotalPrice should take one parameter');
-  // TODO replace this comment with your code
+  console.assert(calculateTotalPrice.length === 1, `Failed: The actual number of parameters is ${calculateTotalPrice.length}`);
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex6-totalCost.js:42]: 🥇

File: 1-JavaScript/Week2/assignment/ex6-totalCost.js [hunk 9]
```diff
@@ -21,22 +21,32 @@ instead!
...
 }
 
 function test2() {
   console.log('\nTest 2: return correct output when passed cartForParty');
-  // TODO replace this comment with your code
+  const expected = 'Total: €23.44';
+  const actual = calculateTotalPrice(cartForParty);
+  console.assert(actual === expected, `Failed: Expected "Total: €23.44", but got: ${actual}`);
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex6-totalCost.js:49]: 👍🏽

File: 1-JavaScript/Week2/assignment/ex7-mindPrivacy.js [hunk 10]
```diff
@@ -30,8 +30,8 @@ const employeeRecords = [
 ];
 
 // ! Function under test
-function filterPrivateData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function filterPrivateData(employees) {
+  return employees.map( ({name, occupation, email}) => ({name, occupation, email}));
```
Comment (DanielGordon1) [inline 1-JavaScript/Week2/assignment/ex7-mindPrivacy.js:34]: 💯

**Review by DanielGordon1 — APPROVED** [review]
Solid work on all the assignments! Good job 💯 

Left a question regarding the `ShoppingCartPure`  assignment because although it works and passes the tests I think one edge case scenario might not be covered yet! 

If you have time/energy you could have a second look. 💪🏽
